test/unit/org/apache/cassandra/net/FramingTest.java

```diff
@@ -271,7 +270,7 @@ public void testRandomLegacy()
     private void testRandomLegacy(int count)
     {
         SecureRandom seeds = new SecureRandom();
-        for (int messagingVersion : new int[] { VERSION_30, VERSION_3014, current_version})
+        for (int messagingVersion : new int[] { VERSION_40 })
```

Comment:
We don't need to loop here, there is only a single value. Also, do we need the `FrameDecoderLegacy` class at all?

Reply:
oh, that's a whole new unravel, give me an hour…

Reply:
done.